### ISSUE TITLE
Unit amount invalid format ".66"

### ISSUE THREAD

**Issue by midoNuminix** [braintree-go/braintree-go#278]
Hello, 

Thanks for the library. When representing a  value `NewDecimal(66,2)` it returns the result as `.66`. As a result, braintree returns invalid format for line item unit amount as it expects the value to be `0.66`. A fix will be doing the following: 

```
// MarshalText outputs a decimal representation of the scaled number
func (d *Decimal) MarshalText() (text []byte, err error) {
	b := new(bytes.Buffer)
	if d.Scale <= 0 {
		b.WriteString(strconv.FormatInt(d.Unscaled, 10))
		b.WriteString(strings.Repeat("0", -d.Scale))
	} else {
		str := strconv.FormatInt(d.Unscaled, 10)

		if len(str) < d.Scale {

			str = strings.Repeat("0", d.Scale) + str
		}

		z := str[:len(str)-d.Scale]
		if z == "" {
			z = "0" // To avoid doing strings.Repeat("0", 1) + str
		}
		b.WriteString(z)
		b.WriteString(".")
		b.WriteString(str[len(str)-d.Scale:])
	}
	return b.Bytes(), nil
}

```

**Comment by leighmcculloch** [comment]
Thanks for raising this behavior @midoNuminix. I've made a change #280 to address the serialization inconsistency of `Decimal`.

**Comment by leighmcculloch** [comment]
This was fixed in #280.
